solidity/contracts/hooks/axelar/AxelarHook.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {IPostDispatchHook} from "../../interfaces/hooks/IPostDispatchHook.sol";
+import {StandardHookMetadata} from "../libs/StandardHookMetadata.sol";
+import {Message} from "../../libs/Message.sol";
+
+//TODO: remove temp intermal import. for testing speed purposes only
+interface IAxelarGateway {
+    function callContract(
+        string calldata destinationChain,
+        string calldata destinationContractAddress,
+        bytes calldata payload
+    ) external;
+}
+
+interface IAxelarGasService {
+    function payNativeGasForContractCall(
+        address sender,
+        string calldata destinationChain,
+        string calldata destinationAddress,
+        bytes calldata payload,
+        address refundAddress
+    ) external payable;
+}
+
+contract AxelarHook is IPostDispatchHook {
+    using StandardHookMetadata for bytes;
+    using Message for bytes;
+
+    IAxelarGasService public immutable AXELAR_GAS_SERVICE;
+    IAxelarGateway public immutable AXELAR_GATEWAY;
+    string public DESTINATION_CHAIN;
+    string public DESTINATION_CONTRACT;
+    bytes GMP_CALL_CODE;
```

Comment:
interesting, would be curious to learn more about what this is

Reply:
we also have some clever tricks for storing immutable bytes elsewhere in the codebase
see `MetaProxy`

Reply:
this is just any data required by the GMP call to get the paylod to the receiver chain ISM. will change it's name